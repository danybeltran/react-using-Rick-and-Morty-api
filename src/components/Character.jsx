import React from "react";
import { CharacterImage, FavoriteButton, InformationTable } from "./Character/index";

export const Character = (props, { isFavorite, test }) => {
    return (
        <>
            <div className="character">
                <div className="character-body">
                    <CharacterImage {...props} />
                    <div className="character-information">
                        <InformationTable {...props} />
                        <br />
                        <FavoriteButton {...{ id: props.id, isFavorite: test ? true : isFavorite }} />
                    </div>
                </div>
            </div>
        </>
    )
}