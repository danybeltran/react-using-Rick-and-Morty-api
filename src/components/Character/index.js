export { CharacterImage }
from "./CharacterImage"
export { FavoriteButton }
from "./FavoriteButton"
export { Information }
from "./Information"
export { InformationTable }
from "./InformationTable"
export { Character }
from "../Character";