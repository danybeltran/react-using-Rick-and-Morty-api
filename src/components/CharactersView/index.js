export { CharactersBody }
from "./CharactersBody";
export { CharactersContainer }
from "./CharactersContainer";
export { CharactersHeader }
from "./CharactersHeader";
export { MappedCharacters }
from "./MappedCharacters";
export { Pagination }
from "./Pagination"
export { PaginationLink }
from "./PaginationLink";
export { Results }
from "./Results"
export { SearchButton }
from "./SearchButton";
export { SearchInput }
from "./SearchInput";
export { ZeroFound }
from "./ZeroFound";