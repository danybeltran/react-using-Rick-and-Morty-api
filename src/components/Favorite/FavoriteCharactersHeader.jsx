import React from "react";

export const FavoriteCharactersHeader = ({ text }) => {
    return <h3 className="characters-header">{text}</h3>
}