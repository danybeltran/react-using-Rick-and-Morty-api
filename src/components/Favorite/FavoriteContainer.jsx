import React from "react";

export const FavoriteContainer = ({ children }) => {
    return (
        <div className="favorite">
            {children}
        </div>
    )
}