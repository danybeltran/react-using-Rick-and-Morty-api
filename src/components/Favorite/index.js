export { FavoriteCharacters } from "./FavoriteCharacters"
export { FavoriteCharactersHeader } from "./FavoriteCharactersHeader"
export { FavoriteContainer } from "./FavoriteContainer"
export { MappedFavoriteCharacters } from "./MappedFavoriteCharacters"
export { NoResults } from "./NoResults"