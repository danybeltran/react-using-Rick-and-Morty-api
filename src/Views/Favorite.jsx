<<<<<<< HEAD
import React, { useState, useEffect } from "react";
import { Character } from "../components/Character";
import MortyFace from "../img/morty-no-favorites.png";

export const Favorite = () => {
    // get the string of all ids separated by comma to fetch the api
    const [AllFavoriteCharactersId] = useState(localStorage["favorite"].replace(/\[|\]/g, ""));

    const [favoriteCharacters, setFavoriteCharacters] = useState([]);


    useEffect(() => {
        const getFavoriteCharacters = async (ids) => {
            var favorite = await fetch("https://rickandmortyapi.com/api/character/" + ids + ",");
            favorite = await favorite.json();
            setFavoriteCharacters(favorite);
        }
        getFavoriteCharacters(AllFavoriteCharactersId);
    },[AllFavoriteCharactersId])
    return (
        <div className="favorite">
            <h3 className="characters-header">Your favorite characters</h3>
            <br />
            <div className="favorite-b">
                {
                    typeof favoriteCharacters.map !== "undefined" ?
                        favoriteCharacters.map((character) => <Character props={character} isFavorite={true} key={character.id} />)
                        :
                        <>
                            <div className="container-no-results">
                                <img src={MortyFace} className="no-results-image" alt="" />
                                <h3 className="no-results-text">List is empty</h3>
                            </div>
                        </>
                }
            </div>
        </div>
    )
=======
import React from "react";
import { FavoriteContainer } from "../components/Favorite/FavoriteContainer";
import { FavoriteCharactersHeader } from "../components/Favorite/FavoriteCharactersHeader";
import { FavoriteCharacters } from "../components/Favorite/FavoriteCharacters";

export const Favorite = () => {
    return (
        <FavoriteContainer>
            <FavoriteCharactersHeader text="Your favorite characters" />
            <br />
            <FavoriteCharacters />
        </FavoriteContainer>
    )
>>>>>>> 32ef5e00d4735d6783d785b48d2330430e388d02
}