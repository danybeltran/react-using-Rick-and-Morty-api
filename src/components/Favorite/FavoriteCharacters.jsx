import React from "react";
import { MappedFavoriteCharacters } from "../Favorite/MappedFavoriteCharacters";
export const FavoriteCharacters = () => {

    return (
        <div className="favorite-b flex">
            <MappedFavoriteCharacters />
        </div>
    )
}