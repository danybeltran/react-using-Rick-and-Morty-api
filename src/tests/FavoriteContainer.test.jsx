import React from "react";
import { FavoriteContainer } from "../components/Favorite/";

import renderer from "react-test-renderer";

test("Should render a 'h1' with text 'A test' and a 'p' with text 'Paragraph'", () => {
    const renderdComponent = renderer.create(
        <FavoriteContainer><h1>A test</h1><p>Paragraph</p></FavoriteContainer>
    );

    let component = renderdComponent.toJSON();

    expect(component).toMatchSnapshot();

    expect


})