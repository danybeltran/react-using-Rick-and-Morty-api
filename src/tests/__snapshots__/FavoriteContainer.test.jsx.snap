// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render a 'h1' with text 'A test' and a 'p' with text 'Paragraph' 1`] = `
<div
  className="favorite"
>
  <h1>
    A test
  </h1>
  <p>
    Paragraph
  </p>
</div>
`;
